### PR TITLE
stat/distuv: add inverse gamma distribution

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: stat/distuv/inversegamma_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"sort"
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestInverseGamma(t *testing.T) {
+	// Values a comparison with scipy
```
Comment (sbinet) [inline stat/distuv/inversegamma_test.go]: `s/Values a comparison/Values extracted from a comparison/`
perhaps ?

Comment (btracey) [reply stat/distuv/inversegamma_test.go]: Done.

File: stat/distuv/inversegamma.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	// TODO(btracey): Replace this with a direct call to the upper regularized
+	// gamma function if mathext gets it.
+	//return 1 - mathext.GammaInc(g.Alpha, g.Beta/x)
+	return mathext.GammaIncComp(g.Alpha, g.Beta/x)
+}
+
+// ExKurtosis returns the excess kurtosis of the distribution.
+func (g InverseGamma) ExKurtosis() float64 {
+	if g.Alpha < 4 {
```
Comment (kortschak) [inline stat/distuv/inversegamma.go]: `<=` to save the work below.

File: stat/distuv/inversegamma.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	a := g.Alpha
+	b := g.Beta
+	lg, _ := math.Lgamma(a)
+	return a*math.Log(b) - lg + (-a-1)*math.Log(x) - b/x
+}
+
+// Mean returns the mean of the probability distribution.
+func (g InverseGamma) Mean() float64 {
+	if g.Alpha < 1 {
```
Comment (kortschak) [inline stat/distuv/inversegamma.go]: `<=`

File: stat/distuv/inversegamma.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+		return math.Inf(1)
+	}
+	return g.Beta / (g.Alpha - 1)
+}
+
+// Mode returns the mode of the distribution.
+func (g InverseGamma) Mode() float64 {
+	if g.Alpha < 1 {
```
Comment (kortschak) [inline stat/distuv/inversegamma.go]: Is this needed here?

File: stat/distuv/inversegamma.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+// Prob computes the value of the probability density function at x.
+func (g InverseGamma) Prob(x float64) float64 {
+	return math.Exp(g.LogProb(x))
+}
+
+// Quantile returns the inverse of the cumulative distribution function.
+func (g InverseGamma) Quantile(p float64) float64 {
+	if p < 0 || p > 1 {
```
Comment (kortschak) [inline stat/distuv/inversegamma.go]: `if p < 0 || 1 < p {`

File: stat/distuv/inversegamma.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+// StdDev returns the standard deviation of the probability distribution.
+func (g InverseGamma) StdDev() float64 {
+	return math.Sqrt(g.Variance())
+}
+
+// Variance returns the variance of the probability distribution.
+func (g InverseGamma) Variance() float64 {
+	if g.Alpha < 2 {
```
Comment (kortschak) [inline stat/distuv/inversegamma.go]: `<=`

**Comment by btracey** [comment]
Updated to work with #477 . PTAL

**Pull request merged**